### PR TITLE
[翻译完成] 20191213 Go: Goroutine and Preemption

### LINKED ISSUE #1641: 20191213 Go: Goroutine and Preemption
标题：Go: Goroutine and Preemption

原文链接：https://medium.com/a-journey-with-go/go-goroutine-and-preemption-d6bc2aa2f4b7

译者自己根据原文翻译，译文保存为 markdown 格式，并附上尾部的签名。

```
---
via: https://medium.com/a-journey-with-go/go-goroutine-and-preemption-d6bc2aa2f4b7

作者：[Vincent Blanchon](https://medium.com/@blanchon.vincent)
译者：[译者ID](https://github.com/译者ID)
校对：[校对者ID](https://github.com/校对者ID)

本文由 [GCTT](https://github.com/studygolang/GCTT) 原创编译，[Go 中文网](https://studygolang.com/) 荣誉推出
```

### PR DESCRIPTION
**Pull request by SarahChenBJ**
close #1641

### CODE REVIEW

**Comment by SarahChenBJ** [comment]
Issue: https://github.com/studygolang/GCTT/issues/1641

File: translated/tech/20191213-Go-Goroutine-and-Preemption.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+}
+```
+
+由于没有函数调用，所有的 goroutine 永远都不会被 block，所以调度器没有抢占这些 goroutine。我们来看一下 tracing中的状态：*（所有的 goroutines 没有被抢占）*
+![STW_goroutines_preemption](https://github.com/SarahChenBJ/gctt-image/blob/master/20191213-goroutin-and-preemption/force_preemption_tracing.png?raw=true)
+<br>
+
+为了解决这个问题，Go 提供了多种的解决办法：
+
+- 强制调度器为`runtime.Gosched()`方法执行 `thanks`:
```
Comment (unknwon) [inline translated/tech/20191213-Go-Goroutine-and-Preemption.md:135]:
```suggestion
- 通过调用 `runtime.Gosched()` 强制调度器进行调度操作：
```

File: translated/tech/20191213-Go-Goroutine-and-Preemption.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+同时这篇文章也对比了几种实现技术的利弊，预计这个“非协作式抢占”会在下一个 Go 版本中实现。
+
+----------------
+
+via: https://medium.com/a-journey-with-go/go-goroutine-and-preemption-d6bc2aa2f4b7
+
+作者：[Vincent Blanchon](https://medium.com/@blanchon.vincent)
+译者：[SarahChenBJ](https://github.com/SarahChenBJ)
+校对：[校对者ID](https://github.com/校对者ID)
```
Comment (unknwon) [inline translated/tech/20191213-Go-Goroutine-and-Preemption.md:185]:
```suggestion
校对：[unknwon](https://github.com/unknwon)
```

**Comment by SarahChenBJ** [comment]
@unknwon all comments are resolved, please review. Thanks.

**Comment by unknwon** [comment]
cc @polaris1119

**Pull request merged**